libkbfs/file_data.go

```diff
@@ -97,6 +122,341 @@ func (fd *fileData) getFileBlockAtOffset(ctx context.Context,
 	return ptr, parentBlocks, block, nextBlockStartOff, startOff, wasDirty, nil
 }
 
+// getLeafBlocksForOffsetRange fetches all the leaf ("direct") blocks
+// that encompass the given offset range (half-inclusive) in the file.
+// If `endOff` is -1, it returns blocks until reaching the end of the
+// file.  Note the range could be made up of holes, indicated in the
```

Comment:
instead of `indicated in the ...`, perhaps you could say something like `...made up of holes, meaning that the last byte of a direct block doesn't immediately precede the first byte of the subsequent block.`

As an aside, I had to re-figure out how holes worked ( https://keybase.slack.com/archives/kbfs/p1471475006002324 ) -- can you put a comment in block_types.go somewhere explaining how it works?